src/main/java/bio/terra/flight/dataset/create/SetBigQueryAuthorization.java

```diff
@@ -0,0 +1,46 @@
+package bio.terra.flight.dataset.create;
+
+import bio.terra.pdao.bigquery.BigQueryContainerInfo;
+import bio.terra.pdao.bigquery.BigQueryPdao;
+import bio.terra.service.JobMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SetBigQueryAuthorization implements Step {
+    private Logger logger = LoggerFactory.getLogger(SetBigQueryAuthorization.class);
+
+    private BigQueryPdao bigQueryPdao;
+
+    public SetBigQueryAuthorization(BigQueryPdao bigQueryPdao) {
+        this.bigQueryPdao = bigQueryPdao;
+    }
+
+    @Override
+    public StepResult doStep(FlightContext context) {
+        FlightMap workingMap = context.getWorkingMap();
+        BigQueryContainerInfo bqInfo = workingMap.get(
+            JobMapKeys.BQ_DATASET_INFO.getKeyName(), BigQueryContainerInfo.class);
+
+        logger.debug(bqInfo.toString());
+        bigQueryPdao.authorizeDatasetViewsForStudies(bqInfo);
+        bigQueryPdao.addReaderGroupToDataset(bqInfo.getBqDatasetId(), bqInfo.getReadersEmail());
```

Comment:
If we fail to add the readers, should we undo all of the create work?
Is there a way to check that we have valid readers before we get here? Perhaps in the SAM step?

Reply:
Checking for valid readers in the SAM step makes sense.

Since adding readers to a dataset is one of the simplest updates we allow, I think not causing a full failure on dataset create makes sense, but I'm not sure how we best communicate the readers error in with the dataset creation success

Reply:
this call is not related to whether the readers are valid or not. that is done in the CreateDatasetAuthzResource step which is the last step. rori is working on that code and i'll test out what sam returns if a user email is not valid.